Logger/BaseLoggerMixins.cs

```diff
@@ -1,7 +1,57 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
+
     public static class BaseLoggerMixins
     {
-        
+
+        public static void Error(this BaseLogger logNum, string message, params object[] paramArray)
+        {
+            string formattedMessage = string.Format(message, paramArray);
+
+            if(logNum is null)
+            {
+                throw new ArgumentNullException(nameof(logNum));
+            }
+
+            logNum.Log(LogLevel.Error, formattedMessage);
+        }
+
+        public static void Warning(this BaseLogger logNum, string message, params object[] paramArray)
```

Comment:
```suggestion
        public static void Warning(this BaseLogger? logNum, string message, params object[] paramArray)
```
You should make your `BaseLogger` as nullable, as we are allowing null values to be passed through. This will prevent you from having to use the `!` operator in your unit tests.